app/models/rubygem.rb

```diff
@@ -53,7 +53,9 @@ def self.name_starts_with(letter)
   end
 
   def self.total_count
-    count_by_sql "SELECT COUNT(*) from (SELECT DISTINCT rubygem_id FROM versions WHERE indexed = true) AS v"
+    Rails.cache.fetch("gem/total_count", expires_in: 6.hours) do
+      count_by_sql "SELECT COUNT(*) from (SELECT DISTINCT rubygem_id FROM versions WHERE indexed = true) AS v"
```

Comment:
this can be written as

```sql
SELECT count(DISTINCT rubygem_id) FROM versions WHERE indexed = TRUE;
```
in AR I think there's possible to use distinct as well

```ruby
Version.where(indexed: true).distinct.count(:rubygem_id)
```

I can check as well if would not be successful getting more effective query as well.

Reply:
JFYI original query was more optimized, but since index = true is majority of entries, it would not use index and plans are almost similar (select distinct is in current database a little faster).

Reply:
updated to use AR. it is a few milliseconds slower:
```
Rubygem.count_by_sql("SELECT COUNT(*) from (SELECT DISTINCT rubygem_id FROM versions WHERE indexed = true) AS v")
  Rubygem Count (523.0ms)  SELECT COUNT(*) from (SELECT DISTINCT rubygem_id FROM versions WHERE indexed = true) AS v
 => 158781 

2.6.5 :003 > Version.where(indexed: true).distinct.count(:rubygem_id)                                                                                                  
                                                                                                                                                                       
   (547.9ms)  SELECT COUNT(DISTINCT "versions"."rubygem_id") FROM "versions" WHERE "versions"."indexed" = $1  [["indexed", true]]                                      
 => 158781 
```

Reply:
20ms shouldn't matter. AR looks cleaner.